### ISSUE TITLE
Remove the OpenAI support for "guided" generation

### ISSUE THREAD

**Issue by rlouf** [outlines-dev/outlines#347]
### What behavior of the library made you think about the improvement?

OpenAI is not compatible with the main offering of the library: guided generation.

### How would you like it to behave?

Not having any OpenAI integration at all.

**Comment by HerrIvan** [comment]
As a non-expert practitioner, I started testing outlines today, and for the first time (after many months of postponing it) gave it a try with HF transformers (once I realized that the regex method did not work with OpenAI). 

On the other hand, outlines also offers very helpful features (prompt Jinja templating, "is_in"), albeit probably those are less unique (see LMQL). 

However myself, I would be happy to use the above features when querying the OpenAI API. Is it your view that, since there are other repositories that also provide elegant prompting, we should use other libraries for that use-case and let outlines focus on the `neuro-symbolic` problem?
